sql/core/src/main/scala/org/apache/spark/sql/execution/LocalTableScan.scala

```diff
@@ -34,9 +34,16 @@ private[sql] case class LocalTableScan(
 
   protected override def doExecute(): RDD[InternalRow] = rdd
 
+  override lazy val accumulators = Map(
+    "numRows" -> sparkContext.internalAccumulator(0L, "number of rows"))
+
   override def executeCollect(): Array[Row] = {
     val converter = CatalystTypeConverters.createToScalaConverter(schema)
-    rows.map(converter(_).asInstanceOf[Row]).toArray
+    val numRows = accumulator[Long]("numRows")
```

Comment:
we need a specialized accumulator in order to avoid object allocation in +=.

@zsxwing can you just quickly hack together a LongAccumulator?